src/Access/TaxonomyDeleteAccessCheck.php

```diff
@@ -0,0 +1,212 @@
+   * control, and if content is assigned to it. If either of these statements
+   * is true, then 'forbidden' will be returned to prevent the term
+   * from being deleted.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Ignored in this method (not doing user-based access).
+   *
+   * @return \Drupal\Core\Access\AccessResultAllowed|\Drupal\Core\Access\AccessResultForbidden
+   *   Returns 'forbidden' if the term is being used for access control.
+   *   Returns 'allowed' if the term is not being used for access control.
+   */
+  public function access(AccountInterface $account) {
+
+    if ($this->isDeleteAllowed()) {
+      return AccessResult::allowed();
+    }
+    return AccessResult::forbidden();
+  }
+
+  /**
+   * Determine if this term may be deleted.
+   *
+   * @return bool
+   *   TRUE if it may be deleted, FALSE otherwise.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function isDeleteAllowed() {
+    /** @var \Drupal\taxonomy\TermInterface $term */
+    $term = $this->getRouteEntity();
+
+    $retval = TRUE;
+
+    if ($term instanceof TermInterface) {
+      $hasAccessControlMembers = $this->doesTermHaveMembers($term);
+      $assigned_content = $this->isAssignedToContent($term);
+
+      /*
+       * If this term does not have users assigned to it for access
+       * control, and the term is not assigned to any pieces of content,
+       * it is OK to delete it.
+       */
+      if ($hasAccessControlMembers || $assigned_content) {
+
+        /** @var \Drupal\Core\Session\AccountProxyInterface $user */
+        $user = \Drupal::currentUser();
+
+        $override_allowed = $user->hasPermission('allow taxonomy term delete');
+
+        if ($assigned_content && !$override_allowed) {
+          drupal_set_message(t("The term %term is being used to tag content and may not be deleted.",
+            ['%term' => $term->getName()]), 'warning');
+          $retval = FALSE;
+        }
+        elseif ($assigned_content) {
+          drupal_set_message(t("The term %term is being used to tag content.",
+            ['%term' => $term->getName()]), 'warning');
+          $retval = TRUE;
+        }
+
+        if ($hasAccessControlMembers) {
+          drupal_set_message(t("The term %term is being used for access control and may not be deleted.",
+           ['%term' => $term->getName()]), 'warning');
+          $retval = FALSE;
+        }
+
+      }
+    }
+
+    return $retval;
+
+  }
+
+  /**
+   * Determines if this term has active members in it.
+   *
+   * @return bool
+   *   TRUE if the term has members, FALSE otherwise.
+   */
+  private function doesTermHaveMembers(TermInterface $term) {
+
+    /** @var array $sections */
+    $sections = $this->getActiveSections($term);
+
+    if (count($sections) > 0) {
```

Comment:
could be 1 line?
```php
return count($sections) > 0
```

Reply:
I only want a return here if there are actually any entities assigned to this section. Otherwise it should continue looking.